node/tests/SharedTests.ts

```diff
@@ -2074,6 +2074,319 @@ export function runBaseTests<Context>(config: {
+    }
+
+    async function zinterWithScoresWithSumAggregation(client:BaseClient) {
+        const key1 = "{testKey}:1-" + uuidv4();
+        const key2 = "{testKey}:2-" + uuidv4();
+
+        const membersScores1 = { one: 1.0, two: 2.0 };
+        const membersScores2 = { one: 1.5, two: 2.5, three: 3.5 };
+
+        expect(await client.zadd(key1, membersScores1)).toEqual(2);
+        expect(await client.zadd(key2, membersScores2)).toEqual(3);
+
+        // Intersection results are aggregated by the SUM score of elements
+        const zinterWithScoresResults = await client.zinterWithScores([key1, key2], "SUM");
+        const expectedMapSum = {
+            one: 2.5,
+            two: 4.5,
+        };
+        expect(zinterWithScoresResults).toEqual(expectedMapSum);
+    }
+
+    async function zinterWithScoresWithWeightsAndAggregation(client:BaseClient) {
+        const key1 = "{testKey}:1-" + uuidv4();
+        const key2 = "{testKey}:2-" + uuidv4();
+
+        const membersScores1 = { one: 1.0, two: 2.0 };
+        const membersScores2 = { one: 1.5, two: 2.5, three: 3.5 };
+
+        expect(await client.zadd(key1, membersScores1)).toEqual(2);
+        expect(await client.zadd(key2, membersScores2)).toEqual(3);
+
+        // Intersection results are aggregated by the SUM score of elements with weights
+        const zinterWithScoresResults = await client.zinterWithScores([[key1, 3], [key2, 2]], "SUM");
+        const expectedMapSum = {
+            one: 6,
+            two: 11,
+        };
+        expect(zinterWithScoresResults).toEqual(expectedMapSum);
+    }
+
+    async function zinterEmptyCases(client: BaseClient) {
+        const key1 = "{testKey}:1-" + uuidv4();
+
+        // Non existing key zinter
+        expect(
+            await client.zinter([
+                key1,
+                "{testKey}-non_existing_key",
+            ]),
+        ).toEqual([]);
+
+        // Non existing key zinterWithScores
+        expect(
+            await client.zinterWithScores([
+                key1,
+                "{testKey}-non_existing_key",
+            ]),
+        ).toEqual({});
+
+        // Empty list check zinter
+        await expect(client.zinter([])).rejects.toThrow();
+
+        // Empty list check zinterWithScores
+        await expect(client.zinterWithScores([])).rejects.toThrow();
+    }
+   
+
+    it.each([ProtocolVersion.RESP2, ProtocolVersion.RESP3])(
+        `zinter test_%p`,
+        async (protocol) => {
+            await runTest(async (client: BaseClient) => {
+                await zinterBasicTest(client);
+                await zinterWithScoresBasicTest(client);
+                await zinterWithScoresWithMaxAggregation(client);
+                await zinterWithScoresWithMinAggregation(client);
+                await zinterWithScoresWithSumAggregation(client);
+                await zinterWithScoresWithWeightsAndAggregation(client);
+                await zinterEmptyCases(client);
```

Comment:
please split this into multiple test cases

Reply:
I don't see the need to separate it into another test that will only have one case, all the rest should remain as one.

Reply:
If first test fails, others won't run.